src/tools/execution-results.h

```diff
@@ -144,7 +144,9 @@ struct ExecutionResults {
   }
 
   bool areEqual(Literal a, Literal b) {
-    if (a.type != b.type) {
+    // We allow nulls to have different types (as they compare equal regardless)
+    // but anything else must have an identical type.
```

Comment:
Should we be checking subtype relations rather than equality? For nulls specifically and for users of this function more generally.

Reply:
Subtype checks would handle nulls properly, but on other values they might not give us what we want.

If the type values are references, we only compare their types (see comment below this code), and if we allow subtyping, we might have one value of `{}` and the other of `{i32}`, say - subtypes, but observably different, and the fuzzer should error.